src/coreclr/System.Private.CoreLib/src/System/Variant.cs

```diff
@@ -297,16 +414,12 @@ CVType switch
                 CV_R8 => BitConverter.Int64BitsToDouble(_data),
                 CV_DATETIME => new DateTime(_data),
                 CV_TIMESPAN => new TimeSpan(_data),
-                CV_ENUM => BoxEnum(),
+                CV_ENUM => Enum.InternalBoxEnum((RuntimeType)_objref!, _data),
```

Comment:
This is end up unreachable, but not directly provable in this PR. I think this PR should focus on OAVariantLib and revert other changes.